linera-chain/src/data_types.rs

```diff
@@ -101,18 +149,44 @@ pub struct ChainAndHeight {
     pub height: BlockHeight,
 }
 
-/// A message received from a block of another chain.
+impl ChainAndHeight {
+    /// Returns the ID of the `index`-th message sent by the block at that height.
+    pub fn to_message_id(&self, index: u32) -> MessageId {
+        MessageId {
+            chain_id: self.chain_id,
+            height: self.height,
+            index,
+        }
+    }
+}
+
+/// A bundle of cross-chain messages.
 #[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize, SimpleObject)]
 pub struct IncomingBundle {
     /// The origin of the message (chain and channel if any).
```

Comment:
Nit: maybe tweak the field documentations in this type to say "messages" or "message bundle". I think we could also remove "The content of" below.